src/messaging/ExchangeMgr.cpp

```diff
@@ -290,7 +290,11 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
             ec = mContextPool.CreateObject(this, payloadHeader.GetExchangeID(), session, false, matchingUMH->Delegate);
         }
 
-        VerifyOrExit(ec != nullptr, err = CHIP_ERROR_NO_MEMORY);
+        if (ec == nullptr)
+        {
+            ChipLogError(ExchangeManager, "OnMessageReceived failed, err = %s", ErrorStr(CHIP_ERROR_NO_MEMORY));
```

Comment:
```suggestion
            ChipLogError(ExchangeManager, "Failed to allocate exchange context to handle received message");
```

Reply:
Same as above

Reply:
Again, should be documented that there's something subtle here.